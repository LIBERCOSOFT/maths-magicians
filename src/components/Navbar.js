import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote',
      text: 'Quote',
    },
  ];

  const activeClassName = 'active__link';

  return (
    <nav>
      <h1>Math Magicians</h1>
      <ul className="nav__links">
        {links.map((link) => (
          <li key={link.id}>
            {' '}
            <NavLink
              to={link.path}
              end
              className={({ isActive }) => (isActive ? activeClassName : null)}
            >
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
