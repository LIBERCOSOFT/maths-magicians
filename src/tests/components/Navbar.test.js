import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../../components/Navbar';

test('renders navbar component snapshot', () => {
  const { asFragment } = render(<Navbar />, { wrapper: BrowserRouter });
  expect(asFragment()).toMatchSnapshot();
});
