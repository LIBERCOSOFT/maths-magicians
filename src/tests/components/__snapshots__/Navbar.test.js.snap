// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar component snapshot 1`] = `
<DocumentFragment>
  <nav>
    <h1>
      Math Magicians
    </h1>
    <ul
      class="nav__links"
    >
      <li>
         
        <a
          aria-current="page"
          class="active__link"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
         
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
         
        <a
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
